stdlib/public/Concurrency/Task.swift

```diff
@@ -46,10 +63,9 @@ public struct Task {
 @available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *)
 extension Task {
 
-  /// Returns 'current' `Task` instance, representing the task from within which
-  /// this function was called.
+  /// Returns the task that this code is being run on.
```

Comment:
I feel we should expand here, how about something like:

```suggestion
  /// Returns the task that this code is being run on.
  /// 
  /// If invoked from the context of an asynchronous function or closure, 
  /// this is guaranteed to return a non-nil task. When called from a synchronous
  /// context the returned value depends on wether the synchronous operation was
  /// itself called from an asynchronous context or not. For example:
  /// 
  ///    func hello() { assert(Task.current != nil) }
  ///    func asynchronous() async { hello() } 
  ///
  /// is correct and will not trigger the assertion, because an asynchronous function exists in the call-stack leading to the `Task.current` invocation.
```

or something like that?